eng/packages/http-client-csharp/generator/Azure.Generator/src/Providers/Abstraction/HttpRequestProvider.cs

```diff
@@ -27,10 +26,10 @@ public override HttpRequestApi FromExpression(ValueExpression original)
             => new HttpRequestProvider(original);
 
         public override InvokeMethodExpression SetHeaders(IReadOnlyList<ValueExpression> arguments)
-            => Original.Property(nameof(PipelineRequest.Headers)).Invoke(nameof(RequestHeaders.Add), arguments);
+            => Original.Property(nameof(PipelineRequest.Headers)).Invoke(nameof(RequestHeaders.SetValue), arguments);
 
         public override AssignmentExpression SetMethod(string httpMethod)
-            => Original.Property(nameof(PipelineRequest.Method)).Assign(New.Instance(typeof(RequestMethod), [Literal(httpMethod)]));
+            => Original.Property(nameof(PipelineRequest.Method)).Assign(Static<RequestMethod>().Invoke(nameof(RequestMethod.Parse), [Literal(httpMethod)]));
```

Comment:
To @annelo-msft's point, we can do the mapping ourselves from the string to the RequestMethod instance to avoid having this calculation occur at runtime.